docs/usage/rtk-query/customizing-queries.mdx

```diff
@@ -406,3 +405,96 @@ const api = createApi({
   }),
 })
 ```
+
+### Adding Meta information to queries
+
+A `baseQuery` can also include a `meta` property in it's return value. This can be beneficial in cases where you may wish to include additional information associated with the request such as a request ID or timestamp.
+
+In such a scenario, the return value would look like so:
+
+1.  ```ts title="Expected success result format with meta" no-transpile
+    return { data: YourData, meta: YourMeta }
+    ```
+2.  ```ts title="Expected error result format with meta" no-transpile
+    return { error: YourError, meta: YourMeta }
+    ```
+
+```ts title="baseQuery example with meta information"
+// file: idGenerator.ts noEmit
+export declare const uuid: () => string
+
+// file: metaBaseQuery.ts
+import {
+  BaseQueryFn,
+  FetchArgs,
+  fetchBaseQuery,
+  FetchBaseQueryError,
+  createApi,
+} from '@reduxjs/toolkit/query'
+import { FetchBaseQueryMeta } from '@reduxjs/toolkit/dist/query/fetchBaseQuery'
+import { uuid } from './idGenerator'
+
+// highlight-start
+type Meta = {
+  requestId: string
+  timestamp: number
+}
+// highlight-end
+
+// highlight-start
+const metaBaseQuery: BaseQueryFn<
+  string | FetchArgs,
+  unknown,
+  FetchBaseQueryError,
+  {},
+  Meta & FetchBaseQueryMeta
+> = async (args, api, extraOptions) => {
+  const requestId = uuid()
+  const timestamp = Date.now()
```

Comment:
requestId and timestamp are a bit unfortunate examples as both of those are already existing - the requestId should even be available here as `api.requestId`.
I can't really think of a better example though :sweat_smile: 
So I guess let's leave it like this now and see once a better example comes up in chat or in an issue :laughing:

Reply:
Yep, I'm definitely not happy with that example but it's all I could think of :sweat_smile: 

It should definitely be replaced when someone comes up with a better usage for custom `meta`

Reply:
Hmm. In the context of the "retry" wrapper.. maybe the amount of retries?

Reply:
Hmm that's a good idea. I'll get some more of your thoughts on it after I have a think on it to try to come up with a feasible example snippet